ote/src/clj/ote/services/pre_notices/operator.clj

```diff
@@ -43,35 +44,49 @@
                  (specql/columns ::transit/pre-notice)
                  {::t-operator/id (op/in (authorization/user-transport-operators db user))})))
 
+(defn- list-attachments-in-db [db user]
+  (specql/fetch db ::transit/pre-notice-attachment
+                (specql/columns ::transit/pre-notice-attachment)
+                {::modification/created-by (authorization/user-id user)}))
+
+(defn- delete-attachments [db config attachments]
+  (doseq [{id ::transit/id} attachments]
+    (specql/delete! db ::transit/pre-notice-attachment {::transit/id id}))
+  (attachment/delete-from-s3 db config attachments))
+
 (defn with-sent-field [notice]
   (if (= (::transit/pre-notice-state notice) :sent)
     (assoc notice ::transit/sent (java.sql.Timestamp. (System/currentTimeMillis)))
     notice))
 
-(defn save-pre-notice [db user notice]
-  (authorization/with-transport-operator-check
-    db user (::t-operator/id notice)
-    (fn []
-      (tx/with-transaction db
-         (let [form-attachments (:attachments notice)
-               n (-> notice
-                     (dissoc :attachments)
-                     (dissoc ::transit/attachments)
-                     (modification/with-modification-fields ::transit/id user)
-                     (with-sent-field))
-               saved-notice (specql/upsert! db ::transit/pre-notice n)
-               notice-id (::transit/id saved-notice)]
-           (log/debug "Save notice: " saved-notice)
-           (specql/update! db ::transit/pre-notice-attachment
-                           {::transit/pre-notice-id notice-id}
-                           (op/and
+(defn save-pre-notice [db user notice config]
+  (let [attachments-from-db (list-attachments-in-db db user)
+        attachments-ids-from-client (set (map ::transit/id (:attachments notice)))
+        to-be-removed (filter #(not (contains? attachments-ids-from-client (::transit/id %)))
```

Comment:
won't this remove all attachments in previous notices by the same user?
`list-attachments-in-db` should only return attachments that *are not* linked yet to a notice?

Reply:
Fixed to list attachment files which are not linked to any pre-notice or files which are linked to current open pre notice.